Formula/cherrytree.rb

```diff
@@ -0,0 +1,29 @@
+class Cherrytree < Formula
+  desc "Hierarchical note taking application featuring rich text and syntax highlighting"
+  homepage "https://www.giuspen.com/cherrytree"
+  url "https://www.giuspen.com/software/cherrytree_0.99.30.tar.xz"
+  sha256 "dff54e8c484beb35531a9aa0987759bc3d7e426979473ada40d1d022fdc50120"
+  license "GPL-3.0-or-later"
+
+  depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+  depends_on "python@3.9" => :build
+  depends_on "adwaita-icon-theme"
+  depends_on "fmt"
+  depends_on "gspell"
+  depends_on "gtksourceviewmm3"
+  depends_on "libxml++"
+  depends_on "spdlog"
+  depends_on "uchardet"
+  uses_from_macos "curl"
+
+  def install
+    system "cmake", ".", "-DBUILD_TESTING=''", *std_cmake_args
+    system "make", "-j4"
+    system "make", "install"
+  end
+
+  test do
+    system "#{bin}/cherrytree", "--version"
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
The unit tests are comprehensive but run in a separate binary at build time, I'm going to install that too and run that if acceptable

Reply:
No, unit tests are not a good fit for the `test` block. Just attempt to interact with the software in some small way a user might, and then check that it behaves as expected.

Reply:
I'm going to install a test document then, will get back

Reply:
One of the existing test fixtures might suit: https://github.com/Homebrew/brew/tree/master/Library/Homebrew/test/support/fixtures

See https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula for details.

Also, you can do something like
```ruby
(testpath/"test.txt").write = "hello"
(testpath/"foo.txt").write <<~EOS
  bar
  qux
EOS
```
In the test block to write a small document to test with.

Reply:
Thanks @carlocab I will propose something in the (european) evening